src/compiler/scala/tools/nsc/settings/ScalaSettings.scala

```diff
@@ -131,7 +132,11 @@ trait ScalaSettings extends AbsScalaSettings
   val sourceReader       = StringSetting       ("-Xsource-reader", "classname", "Specify a custom method for reading source files.", "")
   val reporter           = StringSetting       ("-Xreporter", "classname", "Specify a custom reporter for compiler messages.", "scala.tools.nsc.reporters.ConsoleReporter")
   val strictInference    = BooleanSetting      ("-Xstrict-inference", "Don't infer known-unsound types")
-  val source             = ScalaVersionSetting ("-Xsource", "version", "Treat compiler input as Scala source for the specified version, see scala/bug#8126.", initial = ScalaVersion("2.12"))
+  val source             = ScalaVersionSetting ("-Xsource", "version", "Treat compiler input as Scala source for the specified version, see scala/bug#8126.", initial = ScalaVersion("2.12")).withPostSetHook( setting => {
+    _isScala211 = setting.value >= ScalaVersion("2.11.0")
+    _isScala212 = setting.value >= ScalaVersion("2.12.0")
+    _isScala213 = setting.value >= ScalaVersion("2.13.0")
+  })
```

Comment:
Hmm, given the build failures, I wonder if any of these variables is being set when the `-Xsource` flag is not present.